### ISSUE TITLE
Feature Request: Announce Date for Products

### ISSUE THREAD

**Issue by jacobsb** [istopwg/ippeveselfcert#76]
Hi, This is a request for a new feature so that we can set the "announce date" for each product in the .json file that is created with ippevesubmit.  The products would only be posted on the public PWG website on or after the set announce dates.  This would allow products to be submitted once they are production ready instead of needing to hold the submissions until the products are announced to the public.

**Comment by wifiprintguy** [comment]
@jacobsb We have discussed identical feature requests a number of times over the last couple of years (basically since we moved to the new ippevesubmit system). Because the JSON file is cleartext on the website and loaded into the viewer's browser, there is no way to obfuscate entries. Moreover, the submissions are posted to a publicly available mailing list, and are not private. So even though all printer vendors would appreciate this feature, at this time it isn't practical without a full redesign of the self-certification system. For now, it is up to those registering printers to do so once they have been released to customers to avoid pre-announcing new products.

**Comment by wifiprintguy** [comment]
And +1 from HP, by the way. We also want this enhancement to the IPP Everywhere™ process and concur that this is important. But with my Vice Chair hat on, I know that doing this is much more work than it might appear, even for the old "dynamo" system.

**Comment by michaelrsweet** [comment]
TL;DR: This is a can of worms, and the current member-directed (via submission) method avoids a lot of legal issues.

FWIW, the issue with implementing this functionality has never been technical and adding it to the dynamic portion of the PWG web site would be trivial. Prior to moving to the current JSON file, we had server-side code that validated self-certification results and populated a database, and that database could have been (and still could be) modified to include a publication date, and that could easily be made modifiable by the submitter as needed.

Prior discussion on the [IPP mailing list in 2020](https://www.pwg.org/archives/ipp/2020/020672.html).

Technical requirements to determine:

- Is the "release on date" per-model or per-submission?
- What to do when a product's release is delayed (component shortages, etc.)? I suspect we need to support modifying release dates after the fact, otherwise what member would submit before they were sure of the date(s)?
- What level of confidentiality is required?
- Where are we running this?

Legal:

- Do we need a waiver of liability from members in case they submit something they shouldn't?
- Do we need to require a release date as part of the submission (explicit consent)?
- What confidentiality/IP issues are we going to run into as an organization?
 
Ultimately it comes down to this: PWG members currently control when printers are published on the IPP Everywhere printers page by submitting their results only when they want them public. The advantage of this approach is that we (the PWG) never deal with confidential product information and that it follows the PWG IP policy. The disadvantage is that it puts the onus of timing submissions on the PWG members, and that some models might be delayed beyond when a particular set of certification results might be considered out-of-date (like the transition from 1.0 to 1.1).

Putting the release date logic on the PWG's side will require some careful policy decisions/changes (for handling confidential member information) since the current process and IP policy make the PWG an open organization with non-open discussions limited to the steering committee, and even then the *minutes* of any SC meetings are still published openly... I'm not sure how that will affect officer roles, or whether legal might want to employ separate staff to manage this aspect of the program. We can insulate ourselves somewhat by putting everything in a "secure" database where submitters only have access to their own submissions (what existed before) but ultimately the people maintaining that database and server will have access and will carry a certain amount of liability.

**Comment by michaelrsweet** [comment]
... and to continue the technical discussion, the server-side code could send a confirmation email to the submitter when the release date arrives, at which point they can visit the PWG web site and click "Publish Results" explicitly. Thus, the submission just queues the results and the submitter still needs to explicitly publish them.

So assuming that we *do* implement this, that might be an acceptable way to mitigate any "I didn't mean to publish that" concerns.

**Comment by syoungatcda** [comment]
Canon would like to add their comment to this request as follows:
 
Request: The printer vendor can control when the product information is published on the PWG public web site (IPP Everywhere Certified Printers Page).
 
Currently: Once a test result is posted and the certification result is validated , the product information is immediately published on the IPP-Everywhere Certified Printers Web Page.
 
Expected: Once the certification is validated, the product information is not immediately published. The product information is published when the printer vendor makes a request.  For example:
Plan A. The printer vendor can set a publication date when submitting a test result file to PWG.
Plan B. The printer vendor makes a request (email to PWG) when it’s the day the vendor wants to publish.
 
Current Rule: https://www.pwg.org/archives/ipp/2020/020672.html
 
Background of the request: Printer vendors should not reveal the product information before the sale date.
-          New product information, including product names, is confidential to printer vendors. Therefore, new product information must not be published before the market sale date.
-          Therefore, printer vendors have no choice but to submit IPP-Everywhere test results to the PWG after sale (after market release).
-          In other words, printer vendors cannot use the "IPP-Everywhere logo" or "IPP-Everywhere certified" in their catalogs or their product websites at the time of sale
-          Updating the catalog or in-house product website at a later date after obtaining certification is troublesome and costly twice

**Comment by michaelrsweet** [comment]
@syoungatcda @wifiprintguy I was looking through past F2F meeting minutes for the discussions on this. From section 12 of PWG 5100.20:

> You can use the IPP Everywhere™ logo [LOGO] in marketing material, product packaging, and documentation for products that have passed the IPP Everywhere™ Printer Self- Certification tests and whose results have been submitted to the IEEE-ISTO Printer Working Group. Please contact the IEEE-ISTO Printer Working Group steering committee to obtain written approval for any other usage of the logo.

So basically the current process allows members to request to use the logo prior to submitting their results. I thought for sure we'd discussed this in a prior F2F meeting but I don't see it reflected in the minutes... Anyways, I'll add a slide for the May 2022 F2F to discuss this issue specifically and to suggest that we allow members to proactively start using the logo as soon as a product passes the self-certification tests, with the understanding that the member will submit the results once the product is released.

**Comment by wifiprintguy** [comment]
That sounds good. And we ought to file an erratum against 5100.20 to soften that requirement in its next revision. Not sure we are going to do a new release before v2.0 though.
